packages/block-library/src/site-logo/editor.scss

```diff
@@ -77,9 +80,80 @@
 			justify-content: center;
 		}
 
-		// Hide drag and drop text.
+		// Hide items.
+		.components-placeholder__label,
 		.components-drop-zone__content-text {
 			display: none;
 		}
+
+		// Draw the dashed outline.
+		&::before {
+			content: "";
+			display: block;
+			position: absolute;
+			top: 0;
+			right: 0;
+			bottom: 0;
+			left: 0;
+			border: $border-width dashed currentColor;
+			opacity: 0.3;
+			pointer-events: none;
+
+			// Inherit border radius from style variations.
+			border-radius: inherit;
+		}
+
+		// Style the upload button.
+		.components-placeholder__fieldset {
+			width: auto;
+		}
+
+		color: currentColor;
+		.components-button.components-button {
+			color: inherit;
+			padding: 0;
+			margin: 0;
+			display: flex;
+			justify-content: center;
+			align-items: center;
+			width: $grid-unit-60;
+			height: $grid-unit-60;
+			border-radius: 50%;
+			position: relative;
+			z-index: 1;
```

Comment:
We can avoid z-index if we use natural stacking order. Might need another prop to note the placement of children.